### DISCUSSION TITLE
[Bug Report] @midwayjs/mock 包创建 App 的BUG

### DISCUSSION THREAD

**Discussion by billy-poon** [midwayjs/midway#4241] [General / 闲聊]
用 **mwtsc** + **@midwayjs/mock** 启动开发环境时，如果指定了 `entryFile`，如
```
// package.json
...
"scripts":  {
  "dev": "cross-env NODE_ENV=local mwtsc --watch --run @midwayjs/mock/app.js --entryFile bootstrap.js"
},
...
```

在 **@midway/mock** 包的 *creator.js* 文件 `create` 函数一旦检测到 `entryFile` 选项就会加载对应的文件，然后 `return (undefined);`:

https://github.com/midwayjs/midway/blob/7f84d8558e6784f78b69d035cd236b82176a2093/packages/mock/src/creator.ts#L121

此后在 同一个文件中的 `createApp`  函数并没有对这个返回值进行判断，会导致 `framework.getApplication()` 报 `undefined`: 

https://github.com/midwayjs/midway/blob/7f84d8558e6784f78b69d035cd236b82176a2093/packages/mock/src/creator.ts#L266

然后后续的热重载似乎也需要用到这个返回值，才能正确 `close`。

目前我的临时解决方案是修改安装后的 *creator.js* 的 `createApp` 函数

```
async function createApp(baseDir = process.cwd(), options, customFramework) {
    const framework = await create(baseDir, options, customFramework)
        ?? core_1.getCurrentMainFramework(); // 添加这一行, 其中 core_1 对应 @midwayjs/core 
    return framework.getApplication();
}
```

**Comment by czy88840616** [comment]
用 bootstrap.js 不需要走 @midwayjs/mock，mock 里是给单测用的。
直接起就好

```
 mwtsc --watch --run ./bootstrap.js

```

https://midwayjs.org/docs/deployment#%E6%8C%87%E5%AE%9A%E5%85%A5%E5%8F%A3%E5%90%AF%E5%8A%A8%E6%9C%8D%E5%8A%A1
